connectors/src/connectors/microsoft/index.ts

```diff
@@ -310,8 +319,33 @@ export class MicrosoftConnectorManager extends BaseConnectorManager<null> {
     );
 
     await MicrosoftRootResource.batchMakeNew(newResourcesBlobs);
+    const newNodes = await Promise.all(
+      newResourcesBlobs
+        .filter(
+          (resource) =>
+            resource.nodeType === "folder" || resource.nodeType === "drive"
+        )
+        .map(async (resource) =>
+          itemToMicrosoftNode(
+            resource.nodeType as "folder" | "drive",
+            await getItem(
+              client,
+              typeAndPathFromInternalId(resource.internalId).itemAPIPath
+            )
+          )
+        )
+    );
+    const newNodesWithConnectorId = newNodes.map((nodes) => ({
+      ...nodes,
+      connectorId: connector.id,
+    }))
+    await MicrosoftNodeResource.batchMakeNew(newNodesWithConnectorId);
```

Comment:
This is needed as the `syncFiles` function checks if the `MicrosoftNodeResource` exists.